pkg/queries/user.go

```diff
@@ -347,29 +347,43 @@ func GetUserByID(db *gorm.DB, id uint) (*models.User, error) {
 	return user, nil
 }
 
-// id: 被查询用户的id
-// uid: 查询用户的id
-// ip: 查询用户的ip
-func GetProfile(db *gorm.DB, id uint, uid uint) (models.ProfileResponse, error) {
+func GetProfile(db *gorm.DB, queriedUserID uint, queryingUserID uint) (models.ProfileResponse, error) {
 	if db == nil {
 		db = database.GetDB()
 	}
-	user, err := GetUserByID(db, id)
+
+	user, err := GetUserByID(db, queriedUserID)
 	if err != nil {
 		return models.ProfileResponse{}, err
 	}
+
+	// As written in the previous commit, administrators do not
+	// have access to the invitation code of users other than themselves.
+	var displayInvitationCode bool = queryingUserID == queriedUserID
+	var displayReward bool = queryingUserID == queriedUserID
+
+	if queryingUserID != 0 && queryingUserID != queriedUserID {
```

Comment:
Why check if it's zero? Is it necessary?

Reply:
`GetProfile` should still work when `queryingUserID` is `0` (not logged in). In this case, we need to avoid calling `GetUserByID` with 
 parameter `0`.

Reply:
```suggestion
  // don't query if not logged in
	if queryingUserID != 0 && queryingUserID != queriedUserID {
```